core/src/main/scala/scala/pickling/Macros.scala

```diff
@@ -497,7 +496,7 @@ trait UnpicklerMacros extends Macro with UnpickleMacros with FastTypeTagMacros {
       // in the worst case we can do the same as the interpreted runtime does - just go for allocateInstance
 
       // pending fields are fields that are restored after instantiation (e.g., through field assignments)
-      val pendingFields = if (!canCallCtor) cir.fields else cir.fields.filter(fir =>
+      var pendingFields = if (!canCallCtor) cir.fields else cir.fields.filter(fir =>
```

Comment:
I guess there's no way we can avoid having it be a `var`?

The only reason I ask is because this logic is already kind of hard to follow at first.  I'd love to take this chance to maybe clean up into helper methods that help delineate what we're doing...